adapta/security/clients/aws/_aws_client.py

```diff
@@ -33,8 +34,8 @@ class AwsClient(AuthenticationClient):
     """
 
     def __init__(self, aws_credentials: Optional[AccessKeyCredentials] = None):
-        self._session = None
-        self._credentials = aws_credentials or EnvironmentAwsCredentials()
+        self._session = boto3._get_default_session()
+        self._credentials = aws_credentials or self._session.get_credentials() or EnvironmentAwsCredentials()
```

Comment:
you mix types here? `EnvironmentAwsCredentials` is a type defined in adapta, while `self._session.get_credentials()` is boto3-defined type?

Reply:
I was mixing them indeed, but I don't need it in the end. Actually I intended to remove it but forgot.
It has been removed in [11247c6](https://github.com/SneaksAndData/adapta/pull/437/commits/11247c610a0a01c7dc2a7e504107bd99fcfda553)